components/dashboard/src/data/organizations/orgs-query.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Organization, OrgMemberInfo, User } from "@gitpod/gitpod-protocol";
+import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-mode";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useCallback, useContext } from "react";
+import { useLocation } from "react-router";
+import { publicApiTeamMembersToProtocol, publicApiTeamToProtocol, teamsService } from "../../service/public-api";
+import { getGitpodService } from "../../service/service";
+import { useCurrentUser, UserContext } from "../../user-context";
+
+export interface OrganizationInfo extends Organization {
+    members: OrgMemberInfo[];
+    billingMode?: BillingMode;
+    isOwner: boolean;
+    invitationId?: string;
+}
+
+export function useOrganizationsInvalidator() {
+    const user = useCurrentUser();
+    const queryClient = useQueryClient();
+    return useCallback(() => {
+        console.log("Invalidating orgs... " + JSON.stringify(getQueryKey(user)));
+        queryClient.invalidateQueries(getQueryKey(user));
+    }, [user, queryClient]);
+}
+
+export function useOrganizations() {
+    const user = useCurrentUser();
+    const { refreshUserBillingMode } = useContext(UserContext);
+    const query = useQuery<OrganizationInfo[], Error>(
+        getQueryKey(user),
+        async () => {
+            console.log("Fetching orgs... " + JSON.stringify(getQueryKey(user)));
+            const response = await teamsService.listTeams({});
+            const result: OrganizationInfo[] = [];
+            for (const org of response.teams) {
+                const billingMode = await getGitpodService().server.getBillingModeForTeam(org.id);
+                const members = publicApiTeamMembersToProtocol(org.members || []);
+                const isOwner = members.some((m) => m.role === "owner" && m.userId === user?.id);
+                result.push({
+                    ...publicApiTeamToProtocol(org),
+                    members,
+                    billingMode,
+                    isOwner,
+                    invitationId: org.teamInvitation?.id,
+                });
+            }
+            // refresh user billing mode to update the billing mode in the user context as it depends on the orgs
+            refreshUserBillingMode();
```

Comment:
I've been putting side-effect things like this into an `onSuccess` callback option for queries. I think it provides a little bit of separation where if something breaks in the side-effect call, it doesn't result in the query result being discarded as well. See an example in `components/dashboard/src/data/auth-providers/upsert-org-auth-provider-mutation.ts:26`

Also, I started migrating the user billing mode into react query, so there's a query for it in `components/dashboard/src/data/billing-mode/user-billing-mode-query.ts`. We should probably invalidate that query here as well. Once the couple spots referring to the userBillingMode in the UserContext are updated, we can then drop `refreshUserBillingMode` and just worry about the query invalidation. In short, we could add something like the following query option:

```ts
onSuccess() {
    if(!user) {
        return;
    }
    refreshUserBillingMode();
    queryClient.invalidateQueries(getUserBillingModeQueryKey(user.id));
}
```